crates/autopilot/src/domain/settlement/transaction.rs

```diff
@@ -38,6 +47,20 @@ impl Tx {
     pub fn solver(&self) -> eth::Address {
         self.transaction.solver
     }
+
+    /// Score identical to the one promised during the competition.
+    pub fn check_score(&self) -> bool {
```

Comment:
I don't think the settlement should make any claims about the correctness of the score. IMO having functions `promised_score()` and `observed_score()` and letting the new circuit breaker (or temporary logs) decide itself what to do.